x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -776,7 +776,20 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                         ".metrics-endpoint.metadata_united_default"
                     )
                     .privileges("create_index", "delete_index", "read", "index")
-                    .build(), },
+                    .build(),
+                    // For src/dest indices of the CSP packages that ships a transform
+                    RoleDescriptor.IndicesPrivileges.builder()
+                        .indices("logs-cis_kubernetes_benchmark.findings-*")
+                        .privileges("read", "view_index_metadata")
+                        .build(),
+                    RoleDescriptor.IndicesPrivileges.builder()
+                        .indices(
+                            "cloud_security_posture-findings_latest",
+                            "cloud_security_posture-benchmark_scores"
```

Comment:
Any reason why the names of these indices can't match the [data stream naming scheme](https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme) we use for all other integration indices?

We can still make them hidden, if desired, see https://github.com/elastic/package-spec/issues/266

Reply:
Thanks josh for calling this out
Since these are indices and not datastreams we assumed it would be more fit not to stick with the same naming scheme. And since `latest` transform actually deletes documents from the index and is not append-only kind of index like datastream it can't be a datastream.

I saw this https://github.com/elastic/kibana/issues/121118 as well and wondering what you think is best fit here
Because that raises some questions:
1. what is the type we should use when Transform latest is used?
2. what is the type we should use when Transform pivot is used?
3. how can we support namespaces? and if we should address that with an adhoc solution

Reply:
> Since these are indices and not datastreams we assumed it would be more fit not to stick with the same naming scheme. And since latest transform actually deletes documents from the index and is not append-only kind of index like datastream it can't be a datastream.

Though they aren't data streams, users are going to be granted index access based on index patterns and I think we'd want a pattern to match both the source data streams and the transform destination indices so that users can easily configure access to this data.

> 1\. what is the type we should use when Transform latest is used?

Good question, IMO `logs` makes sense for this use case, and it also matches the source data stream. This could be a good discussion to open up in the issue related to adding official support for `transforms` in the package-spec: elastic/package-spec#23

> 2\. what is the type we should use when Transform pivot is used?

Like above, I think we can match the source data stream's type

> 3\. how can we support namespaces? and if we should address that with an adhoc solution

You'd have to install the transform yourself, but IMO it's probably not worth supporting namespaces in your MVP. I think we should tackle namespace-specific transforms at the Fleet package level and this should be discussed as part of elastic/package-spec#23 as well.

Reply:
thanks a lot josh :) so we will name it
`logs-cloud_security_posture.findings_latest-default` and `logs-cloud_security_posture.benchmark_scores-default`
So that it would fit our planned renaming of our datastreams and future namespace support (currently we would only support default anyway)

Reply:
Hey @joshdover , so we encounter an issue when attempting to create and index that starts with `logs`.

`[2022-03-24T11:56:18.423+02:00][ERROR][plugins.cloudSecurityPosture] illegal_argument_exception: cannot create index with name [logs-cloud_security_posture.findings_latest-default], because it matches with template [logs] that creates data streams only, use create data stream api instead`

since it can't be a datastream, how can we proceed from here? obviously it can't be a datastream because the Transform writes and deletes documents from that index.

Reply:
Hmm shouldn't there be an index template for this index as part of your package? It's important that the index template's priority is high enough to override the default `logs` template

Reply:
@joshdover, thanks for that input, it worked.

@kfirpeled, I change the indices names to:
`logs-cloud_security_posture.findings_latest-default` and `logs-cloud_security_posture.scores-default`